mindsdb/api/mysql/mysql_proxy/executor/executor_wrapper.py

```diff
@@ -0,0 +1,328 @@
+
+        execute_external_router = self.app.route(
+            "/execute_external",
+            methods=[
+                "POST",
+            ],
+        )
+        self.execute_external = execute_external_router(self.execute_external)
+
+        parse_router = self.app.route(
+            "/parse",
+            methods=[
+                "POST",
+            ],
+        )
+        self.parse = parse_router(self.parse)
+
+        do_execute_router = self.app.route(
+            "/do_execute",
+            methods=[
+                "POST",
+            ],
+        )
+        self.do_execute = do_execute_router(self.do_execute)
+        change_default_db_router = self.app.route(
+            "/change_default_db",
+            methods=[
+                "PUT",
+            ],
+        )
+        self.change_default_db = change_default_db_router(self.change_default_db)
+        logger.info(
+            "%s: base params and route have been initialized", self.__class__.__name__
+        )
+
+    def _get_executor(self, params):
+        # We have to send context between client and server
+        # here we load the context json received from the client(mindsdb)
+        # to the local context instance in this Flask thread
+        ctx.load(params["context"])
+        exec_id = params["id"]
+        if exec_id in self.executors_cache:
+            logger.info(
+                "%s: executor %s found in cache", self.__class__.__name__, exec_id
+            )
+            return self.executors_cache[exec_id]
+        session_id = params["session_id"]
+        if session_id in self.sessions_cache:
+            logger.info(
+                "%s: session %s found in cache", self.__class__.__name__, session_id
+            )
+            session = self.sessions_cache[session_id]
+        else:
+            logger.info(
+                "%s: creating new session. id - %s, params - %s",
+                self.__class__.__name__,
+                session_id,
+                params["session"],
+            )
+            session = SessionController()
+            self.sessions_cache[session_id] = session
+        session.database = params["session"]["database"]
+        session.username = params["session"]["username"]
+        session.auth = params["session"]["auth"]
+        session.prepared_stmts = params["session"]["prepared_stmts"]
+        session.packet_sequence_number = params["session"]["packet_sequence_number"]
+        sqlserver = SqlServerStub(connection_id=params["connection_id"])
+
+        logger.info(
+            "%s: session info - id=%s, params=%s",
+            self.__class__.__name__,
+            session_id,
+            session.to_json(),
+        )
+        logger.info(
+            "%s: creating new executor. id - %s, session_id - %s",
+            self.__class__.__name__,
+            exec_id,
+            session_id,
+        )
+        executor = Executor(session, sqlserver)
+        self.executors_cache[exec_id] = executor
```

Comment:
Maybe, but the entry point of 'executor logic' is the Executor class, not ExecutorCommands. Besides splitting logic to client/server in executor commands means that new connection establishes for each new instance of ExecutorCommands which is a performance penalty and also may potentially create some conflicts in case there is a need to create more than 1 ExecutorCommand instance to execute all parts of original command